miden-lib/src/tests/test_prologue.rs

```diff
@@ -304,6 +315,187 @@ pub fn test_prologue_create_account() {
+        transaction.stack_inputs(),
+        MemAdviceProvider::from(transaction.advice_provider_inputs()),
+    );
+    assert!(process.is_ok());
+}
+
+#[cfg_attr(not(feature = "testing"), ignore)]
+#[test]
+pub fn test_prologue_create_account_invalid_fungible_faucet_reserved_slot() {
+    let (account, block_header, chain, notes) = mock_inputs(
+        MockAccountType::FungibleFaucet {
+            acct_id: ACCOUT_ID_FUNGIBLE_FAUCET_INVALID_INITIAL_BALANCE,
+            nonce: ZERO,
+            empty_reserved_slot: false,
+        },
+        AssetPreservationStatus::Preserved,
+    );
+    let code = "
+    use.miden::sat::internal::prologue
+
+    begin
+        exec.prologue::prepare_transaction
+    end
+    ";
+
+    let account_seed: Word = ACCOUNT_SEED_FUNGIBLE_FAUCET_INVALID_INITIAL_BALANCE
+        .iter()
+        .map(|x| Felt::new(*x))
+        .collect::<Vec<_>>()
+        .try_into()
+        .unwrap();
+    let transaction = prepare_transaction(
+        account,
+        Some(account_seed),
+        block_header,
+        chain,
+        notes,
+        code,
+        "",
+        None,
+    );
+
+    let process = run_tx(
+        transaction.tx_program().clone(),
+        transaction.stack_inputs(),
+        MemAdviceProvider::from(transaction.advice_provider_inputs()),
+    );
+    assert!(process.is_err());
+}
+
+#[cfg_attr(not(feature = "testing"), ignore)]
+#[test]
+pub fn test_prologue_create_account_valid_non_fungible_faucet_reserved_slot() {
+    let (account, block_header, chain, notes) = mock_inputs(
+        MockAccountType::NonFungibleFaucet {
+            acct_id: ACCOUNT_ID_NON_FUNGIBLE_FAUCET_VALID_RESERVED_SLOT,
+            nonce: ZERO,
+            empty_reserved_slot: true,
+        },
+        AssetPreservationStatus::Preserved,
+    );
+    let code = "
+    use.miden::sat::internal::prologue
+
+    begin
+        exec.prologue::prepare_transaction
+    end
+    ";
+
+    let account_seed: Word = ACCOUNT_SEED_NON_FUNGIBLE_FAUCET_VALID_RESERVED_SLOT
+        .iter()
+        .map(|x| Felt::new(*x))
+        .collect::<Vec<_>>()
+        .try_into()
+        .unwrap();
+
+    let transaction = prepare_transaction(
+        account,
+        Some(account_seed),
+        block_header,
+        chain,
+        notes,
+        code,
+        "",
+        None,
+    );
+
+    let process = run_tx(
+        transaction.tx_program().clone(),
+        transaction.stack_inputs(),
+        MemAdviceProvider::from(transaction.advice_provider_inputs()),
+    );
+    assert!(process.is_ok())
+}
+
+#[cfg_attr(not(feature = "testing"), ignore)]
+#[test]
+pub fn test_prologue_create_account_invalid_non_fungible_faucet_reserved_slot() {
+    let (account, block_header, chain, notes) = mock_inputs(
+        MockAccountType::NonFungibleFaucet {
+            acct_id: ACCOUNT_ID_NON_FUNGIBLE_FAUCET_INVALID_RESERVED_SLOT,
+            nonce: ZERO,
+            empty_reserved_slot: false,
+        },
```

Comment:
I guess this is the situation where we can't infer the value based on the nonce as suggested in the comments above.

Reply:
Yes we test the failure case for both fungible and non-fungible faucet account creation when the account is created (`nonce == ZERO`) with a non-empty reserved slot.